src/libtexture/imagecache.cpp

```diff
@@ -440,6 +435,19 @@ ImageCacheFile::SubimageInfo::init (const ImageSpec &spec, bool forcefloat)
         if (average_color.size() == size_t(spec.nchannels))
             has_average_color = true;
     }
+
+    if (icfile.m_texformat == TexFormatTexture3d ||
+            icfile.m_texformat == TexFormatShadow ||
+            icfile.m_texformat == TexFormatCubeFaceShadow ||
+            icfile.m_texformat == TexFormatVolumeShadow) {
+        const ParamValue *p = spec.find_attribute ("worldtolocal", TypeMatrix);
+        if (p) {
+            Imath::M44f c2w;
+            icfile.m_imagecache.get_commontoworld (c2w);
+            const Imath::M44f *m = (const Imath::M44f *)p->data();
+            Mlocal.reset (new Imath::M44f (c2w * (*m)));
+        }
+    }
```

Comment:
Slight issue here, the call to `sb.init` from below at line:560 will occur before the call to `ImageCacheFile::init_from_spec`, so icfile.m_texformat has not been set to TexFormatTexture3d yet

Reply:
Oh that looks like it will work, thanks.
If you have a moment tomorrow, maybe can chat?
A few questions/decisions to be made and thinking I'm finally at a point where I won't sound like total buffoon.